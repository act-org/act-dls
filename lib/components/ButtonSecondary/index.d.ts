/**
 * @prettier
 */
import * as React from 'react';
import { ButtonProps } from '@material-ui/core';
export declare type Props = Omit<ButtonProps, 'size'> & {
    size?: 'default' | 'large' | 'small';
};
declare const ButtonSecondary: React.FC<Props>;
export default ButtonSecondary;
//# sourceMappingURL=index.d.ts.map