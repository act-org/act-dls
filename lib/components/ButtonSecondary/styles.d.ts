/**
 * @prettier
 */
declare const _default: (props?: any) => Record<"label" | "default" | "text" | "large" | "small" | "outlined" | "contained", string>;
export default _default;
//# sourceMappingURL=styles.d.ts.map