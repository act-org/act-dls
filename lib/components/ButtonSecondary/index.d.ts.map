{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/ButtonSecondary/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAGH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAC/B,OAAO,EAAU,WAAW,EAAE,MAAM,mBAAmB,CAAC;AAOxD,oBAAY,KAAK,GAAG,IAAI,CAAC,WAAW,EAAE,MAAM,CAAC,GAAG;IAC9C,IAAI,CAAC,EAAE,SAAS,GAAG,OAAO,GAAG,OAAO,CAAC;CACtC,CAAC;AAEF,QAAA,MAAM,eAAe,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CA2BpC,CAAC;AAOF,eAAe,eAAe,CAAC"}