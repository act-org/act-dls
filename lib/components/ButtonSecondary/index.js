"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _Button = _interopRequireDefault(require("@material-ui/core/Button"));

var _clsx = _interopRequireDefault(require("clsx"));

var _mergeClasses = _interopRequireDefault(require("../../helpers/mergeClasses"));

var _styles = _interopRequireDefault(require("./styles"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var ButtonSecondary = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var classesProp = _ref.classes,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'default' : _ref$size,
      otherProps = _objectWithoutProperties(_ref, ["classes", "size"]);

  var classes = (0, _styles["default"])();
  return /*#__PURE__*/React.createElement(_Button["default"], _extends({
    classes: (0, _mergeClasses["default"])({
      contained: (0, _clsx["default"])(classes.contained, classes[size]),
      label: classes.label,
      outlined: (0, _clsx["default"])(classes.outlined, classes[size])
    }, classesProp),
    disableElevation: true,
    ref: ref,
    variant: "outlined"
  }, otherProps));
}); // eslint-disable-next-line immutable/no-mutation

ButtonSecondary.defaultProps = {
  size: 'default'
};
var _default = ButtonSecondary;
exports["default"] = _default;