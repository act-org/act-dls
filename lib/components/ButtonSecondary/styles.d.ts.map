{"version":3,"file":"styles.d.ts","sourceRoot":"","sources":["../../../src/components/ButtonSecondary/styles.ts"],"names":[],"mappings":"AAAA;;GAEG;;AAKH,wBAkCG"}