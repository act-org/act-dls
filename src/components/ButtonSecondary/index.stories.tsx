/**
 * @prettier
 */

/* eslint-disable sort-keys */

import * as React from 'react';
import { action } from '@storybook/addon-actions';
import { boolean, select, text } from '@storybook/addon-knobs';

import Plus from '~/icons/Plus';

import ButtonSecondary from '.';

export const Secondary = (): React.ReactElement<any> => (
  <ButtonSecondary
    disabled={boolean('Disabled?', false)}
    endIcon={boolean('End Icon?', false) ? <Plus /> : undefined}
    onClick={action('onClick()')}
    size={select(
      'Size',
      {
        Small: 'small',
        Default: 'default',
        Large: 'large',
      },
      'default',
    )}
    startIcon={boolean('Start Icon?', true) ? <Plus /> : undefined}
    variant={select(
      'Variant',
      {
        Contained: 'contained',
        Outlined: 'outlined',
        Text: 'text',
      },
      'outlined',
    )}
  >
    {text('Text', 'Click Me')}
  </ButtonSecondary>
);

export default {
  component: ButtonSecondary,
  parameters: {
    info: {
      text: 'This is the secondary variant of the Button component.',
    },
  },
  title: 'Deprecated/Atoms/Button',
};
