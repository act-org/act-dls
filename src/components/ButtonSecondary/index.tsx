/**
 * @prettier
 */
/* eslint-disable react/display-name */

import * as React from 'react';
import { Button, ButtonProps } from '@material-ui/core';
import clsx from 'clsx';
import { get, omit } from 'lodash';

import useStyles from './styles';

export type Props = Omit<ButtonProps, 'size'> & {
  size?: 'default' | 'large' | 'small';
};

const ButtonSecondary: React.FC<Props> = React.forwardRef<
  HTMLButtonElement,
  Props
>(
  (
    { classes: classesProp, size = 'default', ...otherProps }: Props,
    ref,
  ): React.ReactElement<Props> => {
    const classes = useStyles();

    return (
      <Button
        classes={{
          contained: clsx(
            classes.contained,
            classes[size],
            get(classesProp, 'contained'),
          ),
          label: clsx(classes.label, get(classesProp, 'label')),
          outlined: clsx(
            classes.outlined,
            classes[size],
            get(classesProp, 'outlined'),
          ),
          ...(omit(classesProp, ['contained', 'outlined', 'label']) || {}),
        }}
        disableElevation
        ref={ref}
        variant="outlined"
        {...otherProps}
      />
    );
  },
);

// eslint-disable-next-line immutable/no-mutation
ButtonSecondary.defaultProps = {
  size: 'default',
};

export default ButtonSecondary;
