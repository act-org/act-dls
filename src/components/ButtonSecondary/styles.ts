/**
 * @prettier
 */

import { makeStyles } from '@material-ui/core/styles';
import COLORS from '~/constants/colors';

export default makeStyles({
  contained: {
    '&:hover': {
      backgroundColor: COLORS.PRIMARY_DARK,
    },
    backgroundColor: COLORS.PRIMARY,
    borderRadius: 90,
    color: COLORS.WHITE,
  },
  default: {
    maxHeight: 40,
    padding: '8px 16px',
  },
  label: {
    fontSize: 14,
    lineHeight: '24px',
    textTransform: 'none',
  },
  large: {
    maxHeight: 42,
    padding: '9px 16px',
  },
  outlined: {
    border: `solid 2px ${COLORS.PRIMARY}`,
    borderRadius: 90,
    color: COLORS.PRIMARY,
  },
  small: {
    maxHeight: 34,
    padding: '5px 16px',
  },
  text: {
    padding: 0,
  },
});
