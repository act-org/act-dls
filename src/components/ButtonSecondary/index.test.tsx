/**
 * @prettier
 */

import * as React from 'react';
import { noop } from 'lodash';

import render from '~/helpers/test/render';

import ButtonSecondary from '.';

describe('ButtonSecondary', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <ButtonSecondary onClick={noop}>Button Label</ButtonSecondary>,
    );

    expect(container).toMatchSnapshot();
  });
});
