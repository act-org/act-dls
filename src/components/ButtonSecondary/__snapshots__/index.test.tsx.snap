// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSecondary matches the snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-outlined-5 makeStyles-default-2 MuiButton-disableElevation"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label makeStyles-label-3"
    >
      Button Label
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
